purchase.adoc

```diff
@@ -1,174 +1,300 @@
         }
       ],
+      "discounts": [],
       "payments": [
-        {
-          "uuid": "f091c978-bdb0-11e9-9cb5-2a2ae2dbcce4",
-          "amount": 1500,
-          "type": "IZETTLE_CASH",
-          "attributes": {
-            "handedAmount": 10000
-          }
+        ...
         }
       ],
       "cashRegister": {
-        "uuid": "r_dBcr6tT2ypC1jlMuUUzg",
-        "displayName": "Kassa 1"
+        "displayName": "Kassa 2",
+        "uuid": "YdVwSn4iEemdDZepjFuVBw"
       },
       "receiptCopyAllowed": true,
+      "references": {
+        "checkoutUUID": "066e52a6-3589-11eb-a997-5bf729c06b85"
+      },
+      "created": "2020-12-03T17:00:24.879+0000",
+      "refunded": false,
+      "purchaseUUID1": "066e52a6-3589-11eb-a896-5af628c16a84",
+       ...
       "groupedVatAmounts": {
-        "12.0": 1500
+        "12.0": 3500
       },
-      "refund": false,
-      "refunded": false
+      "refund": false
     }
   ],
+  "firstPurchaseHash": "1607015780028Odi3BjWLEeuollr2KMFqhA",
+  "lastPurchaseHash": "1607014824879Bm5SpjWJEeuollr2KMFqhA",
+  "linkUrls": [
+    "<https://purchase.izettle.com/purchases/v2/?limit=2&descending=true&lastPurchaseHash=1607014824879Bm5SpjWJEeuollr2KMFqhA>; rel=\"next\""
+  ]
+}
+----
+--
+
+* Fetching all purchases for a merchant.
++
+--
+*Note:* API consumer should never retrieve all the purchases in one request, since it potentially can put too much load on the server causing request timeout.
+
+Instead, it should request paginated result, using the `limit` and `lastPurchaseHash` parameters.
+
+The `limit` parameter will set the page size, and the `lastPurchaseHash` parameter sets the starting point from where to retrieve purchases.
+Every response will contain a `lastPurchaseHash` attribute, which should added to request for the next page.
+A prepared URL is also provided in attribute `linkUrls`, if more purchases are to be fetched.
+
+In order to load all purchases, begin with an initial request without the `lastPurchaseHash` parameter, this will retrieve the first _n_ purchases , `limit` parameter should be used to define page size.
+
+
+`GET /purchases/v2?limit=50&descending=true`
+
+_Response:_
+
+`Status: 200 OK`
+
+Response will contain the first _n_ number of purchases.
+
+[source,json]
+----
+{
+  "purchases": [
+    ....
+  ],
   "firstPurchaseHash": "14233908814696HbDrnUNRji5iniGikNLiQ",
   "lastPurchaseHash": "1423390928355zj9yI1wyTvqP46AG8NEaYg",
   "linkUrls": [
-        "<https://purchase.izettle.com/purchases/v2?limit=10&descending=true&lastPurchaseHash=1423390928355zj9yI1wyTvqP46AG8NEaYg>; rel=\"next\""
+        "<https://purchase.izettle.com/purchases/v2?limit=50&descending=true&lastPurchaseHash=1423390928355zj9yI1wyTvqP46AG8NEaYg>; rel=\"next\""
   ]
 }
-```
+----
+
+To retrieve the next page, use the value of `linkUrls` in the response.
+
+`GET https://purchase.izettle.com/purchases/v2?limit=50&descending=true&lastPurchaseHash=1423390928355zj9yI1wyTvqP46AG8NEaYg`
```

Comment:
To load all purchases in paginated results:
1. Send a request with the `limit` parameter to define the number of purchases to retrieve. 
    `<add the request example here>`
2. In the bottom of the response, find the values of the `lastPurchaseHash` and `linkUrls` attributes.
`<add the response example here>`
3. Send another request that has the `limit` parameter set the same as in the initial request and the `lastPurchaseHash` and `linkUrls` set with the values from the previous response.
`<add the request example here>`
4. Repeat step 2 and step 3 until the response is empty.

Reply:
Great suggestion! 
Maybe we can even add a code snippet for that 🤔